version: '3.8'

services:
  rabbitmq:
    image: rabbitmq:3-management
    container_name: rabbitmq
    ports:
      - "5672:5672"
      - "15672:15672"
    environment:
      RABBITMQ_DEFAULT_USER: guest
      RABBITMQ_DEFAULT_PASS: guest
    networks:
      - rede-trabalho06
    healthcheck:
      test: ["CMD", "rabbitmq-diagnostics", "check_port_connectivity"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - rabbitmq_data:/var/lib/rabbitmq

  gerador:
    build:
      context: ./gerador_mensagens
    container_name: gerador
    depends_on:
      rabbitmq:
        condition: service_healthy
    networks:
      - rede-trabalho06
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "10m"
        max-file: "3"

  consumidor_face:
    build:
      context: ./consumidor_face
    container_name: consumidor_face
    depends_on:
      rabbitmq:
        condition: service_healthy
    networks:
      - rede-trabalho06
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "10m"
        max-file: "3"

  consumidor_team:
    build:
      context: ./consumidor_team
    container_name: consumidor_team
    depends_on:
      rabbitmq:
        condition: service_healthy
    networks:
      - rede-trabalho06
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "10m"
        max-file: "3"

networks:
  rede-trabalho06:
    driver: bridge

volumes:
  rabbitmq_data:
